src/Article.jsx

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from "react";
+import { GetArticle } from "../utils/api";
+import ArticlePage from "./ArticlePage";
+import { useParams } from "react-router-dom";
+import { Link } from "react-router-dom";
+
+const Article = () => {
+  const { article_id } = useParams();
+  const [article, setArticle] = useState("");
+
+  useEffect(() => {
+    if (article_id) {
+      GetArticle(article_id).then(({ article }) => {
+        setArticle(article);
+      });
+    }
+  }, [article_id]);
```

Comment:
Nice logic, but you do not actually need anything in the dependency array for this. You also dont need an if statement as this should only be rendered if we are on the right path of `/articles/article_id`